entity_properties/defns.ttl

```diff
@@ -0,0 +1,227 @@
+
+prop:hasPhaseCount a  prop:ElectricalProperty ;
+    skos:definition "Entity has given phase count" ;
+    rdfs:range prop:PhaseCountShape .
+
+prop:PowerComplexityShape a sh:NodeShape ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        sh:path prop:value ;
+        sh:in   ("real" "reactive" "apparent") ;
+        sh:minCount 1 ;
+    ] .
+prop:PowerFlowShape  a sh:NodeShape ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        sh:path brick:value ;
+        sh:in   ("import" "export" "net" "absolute") ;
+        sh:minCount 1 ;
+    ] .
+prop:PhasesShape  a sh:NodeShape ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        sh:path brick:value ;
+        sh:in   ("A" "B" "C" "AB" "BC" "AC" "ABC") ;
+        sh:minCount 1 ;
+    ] .
+
+prop:PhaseCountShape    a sh:NodeShape ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        a   sh:PropertyShape ;
+        sh:path brick:value ;
+        sh:datatype xsd:nonNegativeInteger ;
+        sh:minCount 1 ;
+    ] .
+
+
+prop:hasSerialNumber    a   prop:InherentProperty, prop:IdentityProperty ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        a   sh:PropertyShape ;
+        sh:path brick:value ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] .
+
+################################
+### Networking Properties
+################################
+
+prop:NetworkProperty    rdfs:subClassOf prop:VirtualProperty ;
+    skos:definition "Properties related to the networked aspects of an entity" .
+
+prop:hasNetworkAddress  a   prop:NetworkProperty ;
+    skos:definition "Entity has the given network address" .
+
+prop:hasMACAddress  rdfs:subPropertyOf  prop:hasNetworkAddress ;
+    a   prop:InherentProperty ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        a   sh:PropertyShape ;
+        sh:path brick:value ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] .
+
+prop:hasNMI  rdfs:subPropertyOf  prop:hasNetworkAddress ;
+    a   prop:InherentProperty ;
+    rdfs:range qudt:Quantity ;
+    sh:property [
+        a   sh:PropertyShape ;
+        sh:path brick:value ;
+        sh:datatype xsd:string ;
+        sh:minCount 1 ;
+    ] .
+
+prop:hasIPAddress   rdfs:subPropertyOf  prop:hasNetworkAddress ;
+    a   prop:ConfigurableProperty .
+prop:hasIP4Address  rdfs:subPropertyOf  prop:hasIP4Address .
+prop:hasIP6Address  rdfs:subPropertyOf  prop:hasIP4Address .
+
+
+################################
+### Aggregation Properties
+################################
+
+prop:aggregate  a prop:VirtualProperty ;
+    rdfs:range  prop:AggregationShape .
+
+prop:AggregationShape   a   sh:NodeShape ;
+    sh:property [
+        sh:path prop:aggregationFunction ;
+        sh:in   ("max", "min", "mean", "mode", "median", "count", "sum") ;
+        sh:minCount 1 ;
+    ] ;
+    sh:property [
+        # TODO: how to represent durations?
+        sh:path prop:aggregationInterval ;
+        sh:datatype xsd:string ; # e.g. "15m" ?
```

Comment:
Hi Gabe here is what we do in BSync: https://github.com/BuildingSync/schema/blob/4db7102a6c627bade6e5dbb1861ac96633adfb9b/BuildingSync.xsd#L17805